src/types/Criterion.cpp

```diff
@@ -0,0 +1,55 @@
+#include "../../include/types/Criterion.h"
+#include <iostream>
+#include <string>
+
+Criterion::Criterion(std::string id) {
+  id_ = id;
+  name_ = "";
+  direction_ = 1;
+  weight_ = 0.;
+}
+
+Criterion::Criterion(std::string id, std::string name, int direction,
+                     float weight) {
+  id_ = id;
+  name_ = name;
+  direction_ = direction;
+  weight_ = weight;
+}
+
+Criterion::Criterion(const Criterion &crit) {
+  id_ = crit.getId();
+  name_ = crit.getName();
+  direction_ = crit.getDirection();
+  weight_ = crit.getWeight();
+}
+
+std::string Criterion::getId() const { return id_; }
+
+void Criterion::setId(std::string id) { id_ = id; }
+
+std::string Criterion::getName() const { return name_; }
+
+void Criterion::setName(std::string name) { name_ = name; }
+
+int Criterion::getDirection() const { return direction_; }
+
+void Criterion::setDirection(int direction) { direction_ = direction; }
+
```

Comment:
more recent version of Criterion on master maybe we should wait to merge other PR and do this one last

Reply:
Yeah no worries about this, all the modification in Criteria and Criterion was done by my linter, I did not intend to change anything here